src/dynamixel_hardware_interface.cpp

```diff
@@ -144,24 +179,41 @@ namespace dynamixel
     **/
     void DynamixelHardwareInterface::write_joints()
     {
-        std::vector<int> command_int;
-        command_int.resize(_dynamixel_ids.size());
 
-        for(unsigned int i=0; i<_dynamixel_ids.size(); i++)
-        {
-            command_int[i] = (int)(_joint_commands[i]*2048/M_PI) + 2048;
+        for (unsigned int i = 0; i < _dynamixel_servos.size(); i++) {
+            std::map<long long int, std::string>::iterator dynamixel_iterator = _dynamixel_map.find(_dynamixel_servos[i]->id());
+            if (dynamixel_iterator == _dynamixel_map.end())
+                continue;
+            // Sending commands only when needed
+            if (std::abs(_joint_commands[i] - _prev_commands[i]) < std::numeric_limits<double>::epsilon())
+                continue;
+            _prev_commands[i] = _joint_commands[i];
+            try {
+                dynamixel::StatusPacket<dynamixel::protocols::Protocol1> status;
+
+                double goal_pos = _joint_commands[i];
+
+                std::map<long long int, double>::iterator dynamixel_corrections_iterator = _dynamixel_corrections.find(_dynamixel_servos[i]->id());
+                if (dynamixel_corrections_iterator != _dynamixel_corrections.end()) {
```

Comment:
What happens if there is no element in the map corresponding to the ID ? Does it risk to crash our program somewhere ?

Why not use `at` method of maps, since we have C++11 ?